### ISSUE TITLE
Please Make 16 Bit Version accessible outside of the crate!

### ISSUE THREAD

**Issue by geoeo** [image-rs/image#1384]
I would like to be able

Use Gray16Image type in my project

My specific use case for this functionality is 

to load depth images

This is more generally applicable to 

computer vision

## Draft

remove the "(crate)" option from pub(crate)

**Comment by HeroicKatora** [comment]
[That's just a typedef](https://github.com/image-rs/image/blob/c0cc1383ea650a437ff94522d4a5e138515f3734/src/buffer.rs#L1307), you replace their use with `ImageBuffer<Luma<u16>, Vec<u16>>`. That being said, the inconsistency with other typdefs seems like an oversight. I think we didn't finisht the discussion on where they should be exported.

**Comment by geoeo** [comment]
Yes thats what im doing now. But as you said it inconsistent. Would be nice to have it fixed. Thanks!

**Comment by virtualritz** [comment]
+1

I want to encode a 16bit PNG. Unless I miss something I am forced to take `image` as a dependency instead on just `png`.
From `png`:
```
pub enum ColorType {
    Grayscale,
    RGB,
    Indexed,
    GrayscaleAlpha,
    RGBA,
}
```
Whereas `image::ColorType` has what I need – `Rgba16`. Also note the capitalization inconsistencies between `image::ColorType` and `png::ColorType` variants (`Rgba` vs `RGBA`).

A related issue is API ergonomics.
The `PngEncoder::encode()` has no variants for different buffer depths – it only takes `[u8]`. 
In my case the input should be a `[u16]` (In below code `quantized_pixel_data` is a `Vec<u16>`):
```rust
        image::png::PngEncoder::new(&mut buffer)
            .encode(
                // Woot?
                unsafe {
                    &*core::ptr::slice_from_raw_parts(
                        quantized_pixel_data.as_ptr() as *const u8,
                        quantized_pixel_data.len() * 2,
                    )
                },
                width as _,
                height as _,
                image::ColorType::Rgba16,
            )
            .unwrap();
```
But I guess that maybe warrants its own ticket? Question is if this should be opened on this tracker or the one for the `png` crate?

**Comment by virtualritz** [comment]
Edit: I saw `png` has `BitDepth` separate (which is probably a good idea. However, the `write_image_data()` method there has the same issue: it only takes `[u8]` data.

**Comment by HeroicKatora** [comment]
> In my case the input should be a [u16]

Moderator note: Yes, please open separate issues for separate issues. Specifically it doesn't help the `png` crate to have an issue floating around here. I'm going to hide the comments in a day.

**Comment by virtualritz** [comment]
> > In my case the input should be a [u16]
> 
> Moderator note: Yes, please open separate issues for separate issues. Specifically it doesn't help the `png` crate to have an issue floating around here. I'm going to hide the comments in a day.

The issue is still for `image`'s `encode()`, *not* `png`. The latter would still need a third issue for `write_image_data()`.

I'll copy the comments over into a new issue and delete them here.
